src/sessions/builder.rs

```diff
@@ -36,6 +37,7 @@ where
     /// FPS defines the expected update frequency of this session.
     fps: usize,
     sparse_saving: bool,
+    desync_detection: bool,
```

Comment:
Maybe instead of a bool, this should be an enum that includes the checksum interval?